### PR TITLE
Update dependency management

### PR DESCRIPTION
**Pull request by helrond**
- Adds dependabot configs
- Updates pre-commit configs
- separates out top-level requirements to `requirements.in`

### CODE REVIEW

**Review by p-galligan — APPROVED** [review]
This looks good, but I'm a little unsure about why it was necessary to create a migration based on the dependencies update.

What was updated here that required it?

**Comment by helrond** [comment]
I believe it's the updated version of asterism.

**Review by HaSistrunk — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -14,3 +14,7 @@ repos:
   rev: v4.3.0
   hooks:
   - id: isort
+- repo: https://github.com/jazzband/pip-tools
+  rev: 6.0.1
```
Comment (bonniegee) [inline .pre-commit-config.yaml]: I think this can be updated to 6.2.0?

**Review by bonniegee — CHANGES_REQUESTED** [review]

File: requirements.in [hunk 2]
```diff
@@ -0,0 +1,12 @@
+asterism==0.7.2
+Django==3.2.4
```
Comment (HaSistrunk) [inline requirements.in]: Django 3.2.5?

**Review by bonniegee — APPROVED** [review]

**Pull request merged**